localstack/services/moto.py

```diff
@@ -0,0 +1,249 @@
+
+    for url_path, handler in backend.flask_paths.items():
+        # endpoints are annotated as string in werkzeug, but don't have to be
+        url_map.add(Rule(url_path, endpoint=handler))
+
+    return url_map
+
+
+def create_aws_request_context(
+    service_name: str,
+    action: str,
+    parameters: Dict = None,
+    region: str = None,
+    endpoint_url: Optional[str] = None,
+) -> RequestContext:
+    """
+    This is a stripped-down version of what the botocore client does to perform an HTTP request from a client call. A
+    client call looks something like this: boto3.client("sqs").create_queue(QueueName="myqueue"), which will be
+    serialized into an HTTP request. This method does the same, without performing the actual request, and with a
+    more low-level interface. An equivalent call would be
+
+         create_aws_request_context("sqs", "CreateQueue", {"QueueName": "myqueue"})
+
+    :param service_name: the AWS service
+    :param action: the action to invoke
+    :param parameters: the invocation parameters
+    :param region: the region name (default is us-east-1)
+    :param endpoint_url: the endpoint to call (defaults to localstack)
+    :return: a RequestContext object that describes this request
+    """
+    if parameters is None:
+        parameters = {}
+    if region is None:
+        region = config.AWS_REGION_US_EAST_1
+
+    service = load_service(service_name)
+    operation = service.operation_model(action)
+
+    # we re-use botocore internals here to serialize the HTTP request, but don't send it
+    client = aws_stack.connect_to_service(
+        service_name, endpoint_url=endpoint_url, region_name=region
+    )
+    request_context = {
+        "client_region": region,
+        "has_streaming_input": operation.has_streaming_input,
+        "auth_type": operation.auth_type,
+    }
+    request_dict = client._convert_to_request_dict(parameters, operation, context=request_context)
+    aws_request = client._endpoint.create_request(request_dict, operation)
+
+    context = RequestContext()
+    context.service = service
+    context.operation = operation
+    context.region = region
+    context.request = create_http_request(aws_request)
+
+    return context
+
+
+def create_http_request(aws_request: AWSPreparedRequest) -> HttpRequest:
+    # create HttpRequest from AWSRequest
+    split_url = urlsplit(aws_request.url)
+    host = split_url.netloc.split(":")
+    if len(host) == 1:
+        server = (to_str(host[0]), None)
+    elif len(host) == 2:
+        server = (to_str(host[0]), int(host[1]))
+    else:
+        raise ValueError
+
+    # prepare the RequestContext
+    headers = Headers()
+    for k, v in aws_request.headers.items():
+        headers[k] = v
+
+    return HttpRequest(
+        method=aws_request.method,
+        path=split_url.path,
+        query_string=split_url.query,
+        headers=headers,
+        body=aws_request.body,
+        server=server,
```

Comment:
I think it makes perfect sense to pass the `server` here, 👍  we just need to be aware that this may slightly change the semantics for some service calls, as so far we've always been calling moto_server with [`localhost:<port>`](https://github.com/localstack/localstack/blob/3d9c1a6f62d26965a503e089ce074ad52ab6bced/localstack/services/edge.py#L254) as the target host. 

This may become interesting for some special cases, like S3 with host- vs path-based bucket addressing (but potentially may just work out of the box!). In any case, no need to change or look into it now - we'll see how it goes once we get there.